File: member/mall-api/src/main/java/com/laf/mall/api/dao/RaffleDao.java
package com.laf.mall.api.dao;

import com.laf.mall.api.dto.RaffleInfo;
import com.laf.mall.api.dto.RaffleLog;
import com.laf.mall.api.dto.RaffleReward;
import com.laf.mall.api.querycondition.raffle.MyRewardQueryCondition;
import com.laf.mall.api.repository.RaffleRepository;
import com.laf.mall.api.vo.MyRewardInfo;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class RaffleDao {

    @Autowired
    private RaffleRepository repository;

    public RaffleInfo getRaffleInfoByActivityCode(final String activityCode, final int mallId) {
        return repository.selectRaffleInfoByActivityCode(activityCode, mallId);
    }

    public int saveRaffleLog(final RaffleLog log) {
        return repository.insertRaffleLog(log);
    }

    public List<MyRewardInfo> getMyRewardList(MyRewardQueryCondition condition) {
        return repository.selectMyRewardList(condition);
    }

    public int getRaffleLogCount(final int memberId, final int raffleId) {
        return repository.selectRaffleLogCount(memberId, raffleId, false);
    }

    public RaffleReward getRewardById(final int trrId) {
        return repository.selectRewardById(trrId);
    }

    public int editRewardInventory(final int trrId, final int inventory) {
        return repository.updateRewardInventory(trrId, inventory);
    }
}

File: member/manager-app/src/main/java/com/laf/manager/dao/RaffleDao.kt
package com.laf.manager.dao

import com.laf.manager.dto.RaffleInfo
import com.laf.manager.dto.RaffleLog
import com.laf.manager.querycondition.raffle.RaffleFilterCondition
import com.laf.manager.repository.RaffleRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

@Component
class RaffleDao {
    @Autowired
    lateinit var repository: RaffleRepository

    fun saveRaffle(raffleInfo: RaffleInfo) = repository.insertRaffleInfo(raffleInfo)

    fun getRaffleByCode(activityCode: String): RaffleInfo? {
        val raffles = repository.selectRaffleByCode(activityCode)

        if (raffles.isNotEmpty())
            return raffles[0]
        else
            return null
    }

    fun getRaffleById(raffleId: Int): RaffleInfo? {
        val raffles = repository.selectRaffleById(raffleId)

        if (raffles.isNotEmpty())
            return raffles[0]
        else
            return null
    }

    fun updateRaffle(raffleInfo: RaffleInfo) = repository.updateRaffleInfo(raffleInfo)

    fun multipleGetRaffleLogs(it: RaffleFilterCondition): List<RaffleLog> {
        val logs = repository.multipleSelectRaffleLogs(it)

        if (logs.isNotEmpty()) {
            return logs
        } else {
            return ArrayList<RaffleLog>()
        }
    }

    fun multipleGetRaffleLogsCount(it: RaffleFilterCondition): Int {
        val count = repository.multipleSelectRaffleLogsCount(it)
        return count
    }
}